exercise/admin.py

```diff
@@ -23,6 +26,20 @@ class Meta:
 class ExerciseAdmin(admin.ModelAdmin):
     form = ExerciseAdminForm
     list_display = ("id", "title", "description", "length_minutes")
+    readonly_fields = ("copied_from",)
+    change_form_template = "admin/exercise/change_form.html"
+
+    def response_change(self, request, obj):
+        if "_copy_exercise" in request.POST:
+            copied_from = Exercise.objects.get(pk=obj.id)
+            obj.title = obj.title + " Copy"
+            obj.copied_from = copied_from
+            obj.id = None
+            obj.save()
```

Comment:
This currently won't copy the relations so please change this block within the if statement to the following:
```suggestion
            # Create a copy of the Exercise
            original_exercise = Exercise.objects.get(pk=obj.id)
            obj.title = obj.title + " Copy"
            obj.copied_from = original_exercise
            obj.id = None
            obj.save()

            # Add ManyToMany connections
            obj.demographics.add(*original_exercise.demographics.all())
            obj.emails.add(*original_exercise.emails.all())
            obj.files.add(*original_exercise.files.all())
```